src/components/stack/BlockItem.js

```diff
@@ -79,52 +82,65 @@ class BlockItem extends Component {
   handleIncrementBursts() {
     if (this.state.numBursts < 10) {
       this.setState({numBursts: this.state.numBursts + 1});
+      this.handleBurstsUpdate();
     }
   }
 
   handleDecrementBursts() {
     if (this.state.numBursts > 1) {
       this.setState({numBursts: this.state.numBursts - 1});
+      this.handleBurstsUpdate();
     }
   }
 
-  /* need to do a PATCH request to firebase and also update the store */
-  handleClickSaveChanges() {
-    // TODO, meet with Xianhai to discuss this
-    this.setState({focusState: FOCUS_INFO});
-  }
-
   /* Input boxes change handlers */
   handleChangeEditTask(e) {
-    this.setState({editTask: e.target.value, modTitle: e.target.value});
+    this.setState({task: e.target.value});
   }
 
   handleChangeEditDescription(e) {
     this.setState(
-        {editDescription: e.target.value, modDescription: e.target.value},
+        {description: e.target.value},
     );
   }
 
   handleChangeEditDurationWork(e) {
     this.setState(
-        {editDurationWork: e.target.value, modDurationWork: e.target.value},
+        {durationWork: e.target.value},
     );
   }
 
   handleChangeEditDurationBreak(e) {
     this.setState(
-        {editDurationBreak: e.target.value, modDurationBreak: e.target.value},
+        {durationBreak: e.target.value},
     );
   }
 
+  handleBlockUpdate(e) {
+    e.preventDefault();
+    this.setState({focusState: FOCUS_INFO});
+    this.props.dataBlockUpdate({
```

Comment:
Try to pass in only the state that is relevant to updating the block. focusState, hover, etc. should be left out. Also pass the block id as a separate argument. I don't think it needs stackId either.

Reply:
Ah okay for sure. I added only the relevant states to be modifed on firebase now.